src/integTest/kotlin/com/ampnet/auditornode/persistence/repository/GatewayIpfsRepositoryIntegTest.kt

```diff
@@ -0,0 +1,211 @@
+        }
+    }
+
+    @Test
+    fun `must correctly fetch text file from IPFS directory`() {
+        val fileContent = "test content"
+        val fileName = "test-file"
+        var directoryHash = IpfsHash("")
+
+        suppose("some file is stored in IPFS directory") {
+            directoryHash = client.uploadFileToIpfsDirectory(fileName = fileName, fileContent = fileContent)
+        }
+
+        verify("GatewayIpfsRepository implementation is used") {
+            assertThat(repository.javaClass)
+                .isEqualTo(GatewayIpfsRepository::class.java)
+        }
+
+        verify("file is correctly fetched from IPFS") {
+            val result = repository.fetchTextFileFromDirectory(directoryHash = directoryHash, fileName = fileName)
+
+            assertThat(result)
+                .isRightContaining(IpfsTextFile(fileContent))
+        }
+    }
+
+    @Test
+    fun `must return error for non-existent IPFS text file in a directory`() {
+        verify("GatewayIpfsRepository implementation is used") {
+            assertThat(repository.javaClass)
+                .isEqualTo(GatewayIpfsRepository::class.java)
+        }
+
+        verify("error is returned for non-existent IPFS file") {
+            val fileName = "test"
+            val result = repository.fetchTextFileFromDirectory(
+                directoryHash = IntegTestUtils.NON_EXISTENT_IPFS_HASH,
+                fileName = fileName
+            )
+
+            assertThat(result)
+                .isLeftSatisfying {
+                    assertThat(it)
+                        .isInstanceOf(IpfsError.IpfsHttpError::class)
+                }
+        }
+    }
+
+    @Test
+    fun `must correctly fetch binary file from IPFS directory`() {
+        val fileContent = "test content"
+        val fileName = "test-file"
+        var directoryHash = IpfsHash("")
+
+        suppose("some file is stored in IPFS directory") {
+            directoryHash = client.uploadFileToIpfsDirectory(fileName = fileName, fileContent = fileContent)
+        }
+
+        verify("GatewayIpfsRepository implementation is used") {
+            assertThat(repository.javaClass)
+                .isEqualTo(GatewayIpfsRepository::class.java)
+        }
+
+        verify("file is correctly fetched from IPFS") {
+            val result = repository.fetchBinaryFileFromDirectory(directoryHash = directoryHash, fileName = fileName)
+
+            // we must isRightSatisfying + assertThat because ByteArray does not have sensible equals method
+            assertThat(result)
+                .isRightSatisfying {
+                    assertThat(it.content).isEqualTo(fileContent.toByteArray())
+                }
+        }
+    }
+
+    @Test
+    fun `must return error for non-existent IPFS binary file in a directory`() {
+        verify("GatewayIpfsRepository implementation is used") {
+            assertThat(repository.javaClass)
+                .isEqualTo(GatewayIpfsRepository::class.java)
```

Comment:
Would it make sense to put this assert in `@BeforeEach`? `GatewayIpfsRepositoryIntegTest` is called the test class so this assert in before each would look good to me.

Reply:
What do you thing about also naming the annotated method `verify that GatewayIpfsRepository implementation is used`?

Reply:
Even better!